### PR TITLE
Reorder case properties for case schema

### PR DESCRIPTION
**Pull request by mkangia**
## Product Description

For domains with access to Data Dictionary, when creating a new case export, the columns for the export will be ordered based on the ordering of case properties in data dictionary.

When editing existing exports, their original ordering is retained because we allow users to set the order of columns when they create exports & so we won't want to change that order for existing exports.

This change is limited to new case exports only.

## Technical Summary

https://dimagi.atlassian.net/browse/SC-3451

### Safety story

Tested locally.
We are just changing the ordering for only new case exports for domains with access to Data Dictionary, so the impact is to limited users/projects.

### Automated test coverage

Added new tests. Old tests should help verify any changes to existing code which is mostly an addition of an optional argument to existing methods.

### QA Plan

Will go through QA. https://dimagi.atlassian.net/browse/QA-6447

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/export/models/new.py [hunk 1]
```diff
@@ -1764,6 +1766,7 @@ def generate_schema_from_builds(
         force_rebuild=False,
         only_process_current_builds=False,
         task=None,
+        for_new_export_instance=False,
```
Comment (zandre-eng) [inline corehq/apps/export/models/new.py:1769]: Should this param also be added to the `generate_schema_from_builds()` function in the `SMSExportDataSchema` class? I'm not entirely sure if this class also gets used in the export view.

**Review by zandre-eng — COMMENTED** [review]
Looks good. Just a few minor comments from my end.

File: corehq/apps/export/models/new.py [hunk 2]
```diff
@@ -1779,6 +1782,7 @@ def generate_schema_from_builds(
             be present in the schema since past builds have not been
             processed.
         :param task: A celery task to update the progress of the build
+        :param for_new_export_instance: Flag to be set if generating schema for a new export instance
```
[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/apps/export/models/new.py:1785]: Ya, I was thinking of that originally. Though that assumption will not always hold.
Like for this use of this function, https://github.com/dimagi/commcare-hq/blob/902a3de50f0538a8a5d84c9e11f632b1b6711a9f/corehq/apps/export/models/new.py#L1841, a fresh schema is being passed to it. Hence, I kept it really explicit so that we only do the reordering when needed.

File: corehq/apps/export/models/new.py (hunk 1 above)
Comment (mkangia) [reply corehq/apps/export/models/new.py:1769]: Absolutely, good catch.
https://github.com/dimagi/commcare-hq/pull/34502/commits/902a3de50f0538a8a5d84c9e11f632b1b6711a9f

Comment (zandre-eng) [reply corehq/apps/export/models/new.py:1769]: Cool, we might also have to include the variable [here](https://github.com/dimagi/commcare-hq/blob/a96f2094c8ffc5f2258eadfa05c133f79d5461ba/corehq/apps/export/views/new.py#L295) so that new bulk case exports also receive the same Data Dictionary ordering.

Comment (mkangia) [reply corehq/apps/export/models/new.py:1769]: I have actually left bulk export out of scope for this PR.
Two reasons:
1. When we do a bulk export, the user is not shown the columns for the export on the UI.
2. For a bulk export, it seems, we [don't create a schema](https://github.com/dimagi/commcare-hq/blob/6b46f48241b95b8a7aff7a42ef467bdb8d946497/corehq/apps/export/views/new.py#L434) only.

I am thinking for bulk export, the order of case properties will not be relevant since there will be so many.
I am happy to add a ticket if you think we should address this.

**Comment by mkangia** [comment]
Hey @esoergel 

Just FYI on how this is looking, feel free to share a review if you feel like.

File: corehq/apps/export/models/new.py (hunk 1 above)
Comment (zandre-eng) [reply corehq/apps/export/models/new.py:1769]: Thanks for sharing your reasoning. I can definitely see it being an okay outcome that users can't set the ordering for bulk case exports (since they can't access the ordering in the UI). Still, I think a ticket could still be a good idea just so that the team can have a chance to discuss this and determine whether there is any value in adding this for bulk case exports.

Comment (mkangia) [reply corehq/apps/export/models/new.py:1769]: Cool, I will add a ticket.

**Comment by mkangia** [comment]
Hey @zandre-eng 

Is it good to go from your end? Any blocking feedback?

**Review by zandre-eng — APPROVED** [review]
> Is it good to go from your end? Any blocking feedback?

Thanks @mkangia for addressing my questions. All looks good from my side :rocket:

**Comment by mkangia** [comment]
Thanks @zandre-eng 

Just pinging this again for review. I missed considering the ordering of case property groups, so I had to revisit my approach a bit.
This now works well. Requesting for another pass for the new commits added.

**Comment by mkangia** [comment]
Also, asking QA for another pass. Also will ask them to add using groups to the test cases.

**Review by zandre-eng — APPROVED** [review]
Good catch on the group indices, PR looks good.

**Comment by mkangia** [comment]
Merging this now. 
QA mentioned [an edge case](https://dimagi.atlassian.net/browse/QA-6489) that will be taken as a follow up.
